### DISCUSSION TITLE
Plugin creation tutorials

### DISCUSSION THREAD

**Discussion by lepigocher** [maplibre/maplibre#75] [Ideas]
Hi,

It could be really usefull to have a - detailled - step-by-step - tutorial about how to configure a project to create a new plugin.
The current documentation from Maplibre and Mapbox on this subject is really the bare minimum!
I have been able to found the one from @astrix and to follow it thanks to Google translator ;-)
Moreover, it could be great to have one using Typescript.

Thanks!

**Comment by HarelM** [comment]
Plugins can have a ton of different forms and use cases, so I think it might be problematic to write a guide that fits all. But feel free to send a PR to improve the docs or checkout awesome-maplibre for other projects and code examples... :-)

**Reply by lepigocher** [reply]
Off course, it's impossible to cover every use cases! :)

But it could be usefull to explain what's recommended in terms of tools (Visual Studio Code (GL-JS), Android Editor (Android Native) and so on). What is required to compile a basic project (the 'classic' toolchain for each environment).
What is the recommended project structure. This could help for newcomers like me to find same kind of file at the same place (assets, src, js, kt, ...).

I looked at several plugins for Maplibre/Mapbox here on Github and found several different project organizations. I understand than you can't force anyone to use a structure but it can help for wider adoption when basic recommendations exists. No need to write a full book for such recommendations, a few pages for each environment (GL JS, Android Native, iOS Native) should be enough.

But it's just an idea! :-)
